### PR TITLE
Fixing bench worker listener registration for windows

### PR DESCRIPTION
**Pull request by simpsont-oci**

### CODE REVIEW

File: performance-tests/bench_2/builder/ListenerFactory.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
+#include "ListenerFactory.h"
+
+namespace Builder {
+
+Bench_Builder_Export
+void RegisterDomainParticipantListener(const std::string& name, const std::function<DDS::DomainParticipantListener::_var_type(const Builder::PropertySeq&)>& factory)
```
Comment (iguessthislldo) [inline performance-tests/bench_2/builder/ListenerFactory.cpp]: Why are these function names capitalized?

**Comment by iguessthislldo** [comment]
Could you explain why this was needed for Windows?

File: performance-tests/bench_2/builder/ListenerFactory.cpp (hunk 1 above)
Comment (simpsont-oci) [reply performance-tests/bench_2/builder/ListenerFactory.cpp]: Hmm, they started life as non-templated classes that inherited from the appropriate template specialization, but then I changed my mind and just made them functions. I'll rename them to use snake case.

**Comment by simpsont-oci** [comment]
We needed some kind of workaround for windows because visual studio complained about the static member variables of an exported template class (For example, VS2015 says, "Builder::ListenerFactory<TopicListener>::s_factory_map: definition of dllimport static data member not allowed" when trying to build Bench_Worker). For the release, we made it compile, but unfortunately as it was it couldn't run on windows because both Bench_Worker and Builder would each define static listener factory maps... the worker would register the type but the builder library wouldn't be able to 'find' it in when trying to call the factory.

**Pull request merged**